Add second E2E gateway failover scenario

tpantelis opened the pull request:
Two gateway nodes configured with 1 submariner engine replica.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/submariner-io/submariner/218)
<!-- Reviewable:end -->

tpantelis commented:
> There are arguably too many implementation details in the test here, but it depends on the implementation of the engine pod anyway, so...

I assume you specifically mean waiting for a new pod and Endpoint instance. I agree that makes it a bit whitebox but I did that to eliminate possible timing issue where, after deleting the pod, we actually run the connectivity test against the previous engine instance before k8s has a chance to react to stop the process/container etc. Perhaps I don't need to check for new  a Endpoint instance but I mainly did that as a sanity check that a new engine instance is running. It also gives it a bit more of a cushion to avoid premature timeout in the connectivity test.

skitt commented:
> I assume you specifically mean waiting for a new pod and Endpoint instance. I agree that makes it a bit whitebox but I did that to eliminate possible timing issue where, after deleting the pod, we actually run the connectivity test against the previous engine instance before k8s has a chance to react to stop the process/container etc. Perhaps I don't need to check for new a Endpoint instance but I mainly did that as a sanity check that a new engine instance is running. It also gives it a bit more of a cushion to avoid premature timeout in the connectivity test.

Right, that’s what I was referring to, but I don’t think it’s much of a problem really. However the fact that it’s useful for timing doesn’t appear in the code; it might be worth adding a comment to that effect (so that in future, if the details change and someone fixes the test, they don’t spend ages figuring out what the point was).

tpantelis commented:
> > I assume you specifically mean waiting for a new pod and Endpoint instance. I agree that makes it a bit whitebox but I did that to eliminate possible timing issue where, after deleting the pod, we actually run the connectivity test against the previous engine instance before k8s has a chance to react to stop the process/container etc. Perhaps I don't need to check for new a Endpoint instance but I mainly did that as a sanity check that a new engine instance is running. It also gives it a bit more of a cushion to avoid premature timeout in the connectivity test.
> 
> Right, that’s what I was referring to, but I don’t think it’s much of a problem really. However the fact that it’s useful for timing doesn’t appear in the code; it might be worth adding a comment to that effect (so that in future, if the details change and someone fixes the test, they don’t spend ages figuring out what the point was).

Added comments and rebased.
